### ISSUE TITLE
Windows: fails to link

### ISSUE THREAD

**Issue by certik** [MarDiehl/stdlib_os#5]
There [is](https://github.com/MarDiehl/stdlib_os/pull/3/checks?check_run_id=1415577537) an error on Windows during linking:
```
[100%] Linking Fortran executable stdlib_test.exe
CMakeFiles\stdlib_test.dir/objects.a(os_path.f90.obj):os_path.f90:(.text+0x4b3): undefined reference to `__os_path_MOD_trim_sep'
collect2.exe: error: ld returned 1 exit status
```

**Comment by MarDiehl** [comment]
@certik: many thanks. I won't have time today. Maybe @arjenmarkus can have a look. He know anyways better what's going on with Windows

**Comment by arjenmarkus** [comment]
@Ondřej Čertík <ondrej@certik.us>  @Martin Diehl <martin.diehl@kuleuven.be>,
I will have a look at this. I know there is a problem with MInGW but that
is of a completely different nature. This one is new to me.

Op wo 18 nov. 2020 om 06:45 schreef Martin Diehl <notifications@github.com>:

**Comment by arjenmarkus** [comment]
I can reproduce this error: it occurs with the MinGW platform. It seems
that the function trim_sep, defined in the module os_path, is not visible
in the submodule os_path_windows of os_path. That function is declared as
private, but does that mean it should not be available to submodules? The
Intel Fortran compiler has no problems with it.

As an experiment I made it public and now the linker is satisfied. I wonder
... Do you know which compiler is correct? My guess is Intel Fortran, as I
see no reasons for a module to restrict access to its contents from its
submodules.

Op wo 18 nov. 2020 om 09:07 schreef Arjen Markus <arjen.markus895@gmail.com

**Comment by MarDiehl** [comment]
I would also say that this is a bug in Gfortran. Submodules should have access to everything in the parent module.

**Comment by arjenmarkus** [comment]
I will create a small reproducer and report it to the Gfortran people. For
now I guess making trim_spec public could be a workaround, but an
alternative is to put a local copy in the submodule. While this causes code
duplication, I am not happy with the "leakage" we have otherwise, What do
you think?

Op vr 20 nov. 2020 om 09:45 schreef Martin Diehl <notifications@github.com>:

**Comment by MarDiehl** [comment]
Let's hope that this bug in gfortran gets fixed soon. For the moment, we simply need to have one of the workarounds, best with a comment why we don't use the best design. Both fixes are ok for me. They are not nice, but also no really a problem.

**Comment by arjenmarkus** [comment]
Right, in that case I will go for the "public" one, as that is merely an
annoyance - if we duplicate the code, an unexpected discrepancy might creep
in at some point.

Op vr 20 nov. 2020 om 12:23 schreef Martin Diehl <notifications@github.com>:

**Comment by arjenmarkus** [comment]
Solved this by making the function trim_sep public. This is very probably a bug in the gfortran compiler.
